### PR TITLE
Move environment init to netplugin-init container

### PR DESCRIPTION
**Pull request by tiewei**
Move envrionment init codes to netplugin-init container.
Make get k8s configs from "/var/contiv/config/contiv.json" to avoid
copying configs around

Signed-off-by: Wei Tie <wtie@cisco.com>

### CODE REVIEW

**Comment by tiewei** [comment]
build PR

File: install/k8s/contiv/contiv.yaml [hunk 1]
```diff
@@ -1,4 +1,99 @@
 ---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
```
Comment (vhosakot) [inline install/k8s/contiv/contiv.yaml]: Contiv works with the default k8s ClusterRoleBindings.  Curious why this is needed? for RBAC?

File: install/k8s/contiv/contiv.yaml [hunk 2]
```diff
@@ -1,4 +1,99 @@
...
+    verbs:
+      - watch
+      - list
+      - update
+      - get
+
+---
+
+apiVersion: v1
+kind: ServiceAccount
```
Comment (vhosakot) [inline install/k8s/contiv/contiv.yaml]: Same question here too. Contiv works with the default `ServiceAccount` cert in (`/var/run/secrets/kubernetes.io/serviceaccount/ca.crt`). Why is this needed?

https://github.com/contiv/netplugin/blob/master/Makefile#L18

Comment (tiewei) [reply install/k8s/contiv/contiv.yaml]: That's CA, not token, I think token is the one indicates k8s role and accounts

File: install/k8s/contiv/contiv.yaml (hunk 1 above)
Comment (tiewei) [reply install/k8s/contiv/contiv.yaml]: RBAC

File: install/k8s/contiv/contiv.yaml (hunk 2 above)
Comment (vhosakot) [reply install/k8s/contiv/contiv.yaml]: Every `ServiceAccount` translates to a `ca.crt` file on disk:

https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/

File: install/k8s/contiv/contiv.yaml [hunk 3]
```diff
@@ -90,20 +143,63 @@ spec:
...
       hostNetwork: true
       hostPID: true
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        effect: NoSchedule
+      serviceAccountName: contiv-netplugin
+      initContainers:
+        - name: contiv-netplugin-init
+          image: contiv/netplugin-etcd-init:latest
+          imagePullPolicy: Always
```
Comment (dseevr) [inline install/k8s/contiv/contiv.yaml]: Could this result in an incompatible init container being pulled down and run with an older release of Contiv?

File: install/k8s/contiv/contiv.yaml [hunk 4]
```diff
@@ -219,21 +289,51 @@ spec:
...
+      - key: node-role.kubernetes.io/master
+        effect: NoSchedule
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      serviceAccountName: contiv-netmaster
       containers:
+      initContainers:
+        - name: contiv-netplugin-init
+          image: contiv/netplugin-etcd-init:latest
+          imagePullPolicy: Always
```
Comment (dseevr) [inline install/k8s/contiv/contiv.yaml]: here, too

File: install/k8s/contiv/contiv.yaml [hunk 5]
```diff
@@ -219,21 +289,51 @@ spec:
...
+                configMapKeyRef:
+                  name: contiv-config
+                  key: contiv_k8s_config
+            - name: CONTIV_CNI_CONFIG
+              valueFrom:
+                configMapKeyRef:
+                  name: contiv-config
+                  key: contiv_cni_config
+          volumeMounts:
+            - mountPath: /var/contiv
```
Comment (dseevr) [inline install/k8s/contiv/contiv.yaml]: Is this for backwards compatibility with the old location?

File: install/k8s/contiv/contiv.yaml [hunk 6]
```diff
@@ -262,52 +357,142 @@ spec:
...
+        - name: contiv-api-proxy
+          image: contiv/auth_proxy:1.2.0
+          args:
+            - --tls-key-file=/var/contiv/auth_proxy_key.pem
+            - --tls-certificate=/var/contiv/auth_proxy_cert.pem
+            - --data-store-address=$(STORE_URL)
+            - --data-store-driver=$(STORE_DRIVER)
+            - --netmaster-address=localhost:9999
+          env:
+            - name: NO_NETMASTER_STARTUP_CHECK
```
Comment (dseevr) [inline install/k8s/contiv/contiv.yaml]: We shouldn't have this env var set by default... this allows auth_proxy to send requests to an incompatible netmaster

File: install/k8s/contiv/contiv.yaml (hunk 3 above)
Comment (tiewei) [reply install/k8s/contiv/contiv.yaml]: it could be, but we have to do it before we starting tagging `netplugin-etcd-init`

**Comment by tiewei** [comment]
close pull request  need to rework based on #1106
